src/ert/storage/ensure_correct_xr_coordinate_order.py

```diff
@@ -0,0 +1,24 @@
+import xarray as xr
+
+
+def ensure_correct_coordinate_order(ds: xr.Dataset) -> xr.Dataset:
```

Comment:
I don't understand this. Do you have an example I can play around with?

Reply:
TLDR; `.to_dataframe()` doesn't "preserve" the index, giving a a dataframe with this index
```
>          new_combined_ds.to_dataframe().index
>          names=['name', 'index', 'realization', 'report_step'], length=500)
```
instead of this
```
>           ensure_correct_coordinate_order(new_combined_ds).to_dataframe().index
>           names=['realization', 'name', 'report_step', 'index'], length=500)

```,
it was already done in `local_ensemble.py` but was also needed for observations so it was moved to a separate file.

This can be tested on main, if you set a breakpoint on 
L1334 in `local_ensemble.py`:
```                
new_combined_ds = ensure_correct_coordinate_order(new_combined_ds)
```
then run
`tests/integration_tests/analysis/test_es_update.py test_that_posterior_has_lower_variance_than_prior`,

When it breakpoints, if you inspect in the python terminal:
```
new_combined_ds
```
it gives this:
```
<xarray.Dataset> Size: 4kB
Dimensions:      (name: 1, index: 10, realization: 50, report_step: 1)
Coordinates:
  * name         (name) object 8B 'POLY_RES'
  * index        (index) int32 40B 0 1 2 3 4 5 6 7 8 9
  * realization  (realization) int32 200B 1 2 3 4 5 6 7 ... 44 45 46 47 48 49 50
  * report_step  (report_step) int32 4B 0
Data variables:
    values       (realization, name, report_step, index) float64 4kB 4.324 .....
```

where `values` seemingly has the dimension ` (realization, name, report_step, index)`, this is true for xarray, but if you do a `.to_dataframe()`, you'd expect the index to remain the same, but doing it gives this:
```
                                           values
name     index realization report_step           
POLY_RES 0     1           0             4.324111
               2           0             0.212733
               3           0             4.134436
               4           0             2.696731
               5           0             0.390561
...                                           ...
         9     46          0            43.097676
               47          0            77.599153
               48          0            86.139757
               49          0            46.357514
               50          0            46.666562
[500 rows x 1 columns]
```
Now name, index is ahead of realization, with report_step at the end.

So that function is a workaround this jumbling up that happens for ...reasons. IF you do a `.[...dims]`, select from the xr dataset in exactly that way, the `.to_dataframe()` preserves the order, hence:
```
ensure_correct_coordinate_order(new_combined_ds).to_dataframe()

```
gives this
```
                                           values
realization name     report_step index           
1           POLY_RES 0           0       4.324111
                                 1       5.775136
                                 2       8.519636
                                 3      12.557611
                                 4      17.889061
...                                           ...
50          POLY_RES 0           5      16.626644
                                 6      22.691176
                                 7      29.719339
                                 8      37.711135
                                 9      46.666562
[500 rows x 1 columns]

```

On the PR, running `test_that_observation_getters_from_experiment_match_expected_data` , will fail if you remove the `ensure_correct_coordinate_order` around the returned single obs dataset. I think something with doing the selecting and dropping messes up the order